experimental/metal/metal_device.m

```diff
@@ -338,81 +337,61 @@ static iree_status_t iree_hal_metal_device_queue_execute(
   iree_hal_metal_device_t* device = iree_hal_metal_device_cast(base_device);
   IREE_TRACE_ZONE_BEGIN(z0);
 
+  iree_hal_resource_set_t* resource_set;
+  IREE_RETURN_AND_END_ZONE_IF_ERROR(
+      z0, iree_hal_resource_set_allocate(&device->block_pool, &resource_set));
+
+  IREE_RETURN_AND_END_ZONE_IF_ERROR(
```

Comment:
at this point if this function failed the resource_set would be leaked - you could do the normal status_is_ok chain or have the resource_set allocate and cleanup be in here and then call another function for actually doing the submission construction (like `resource_set_allocate, status = queue_submit(...); if (!iree_status_is_ok(status)) resource_set_free`)

Reply:
Ah, good catch. Makes sense. Fixed.